src/main/java/seedu/address/logic/commands/DeleteTagPersonCommand.java

```diff
@@ -0,0 +1,118 @@
+
+/**
+ * Deletes a tag from a person identified using its displayed index from the address book.
+ */
+public class DeleteTagPersonCommand extends Command {
+
+    public static final String COMMAND_WORD = "deleteTagPerson";
+    public static final String SHORTENED_COMMAND_WORD = "dtagp";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + " (alias: " + SHORTENED_COMMAND_WORD + ")"
+            + ": Deletes one or more tags from a person identified using its displayed index from the address book.\n"
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_TAG + "catering "
+            + PREFIX_TAG + "budget";
+
+    public static final String MESSAGE_DELETE_TAG_PERSON_SUCCESS = "Deleted tag(s) from person: %1$s.\n";
+    public static final String MESSAGE_DELETE_TAG_PERSON_MISSING_TAGS = "These tags did not seem to exist: %1$s.\n";
+
+    private final Index targetIndex;
+    private final Set<Tag> tagsToDelete;
+
+    /**
+     * @param index of the person in the filtered person list to edit
+     * @param tagsToDelete tags to delete from the person
+     */
+    public DeleteTagPersonCommand(Index targetIndex, Set<Tag> tagsToDelete) {
+        this.targetIndex = targetIndex;
+        this.tagsToDelete = tagsToDelete;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToEdit = lastShownList.get(targetIndex.getZeroBased());
+        Person editedPerson = createEditedPerson(personToEdit, tagsToDelete);
+        model.setPerson(personToEdit, editedPerson);
+        return new CommandResult(formatResultMessage(personToEdit, editedPerson, tagsToDelete));
+    }
+
+    private static Person createEditedPerson(Person personToEdit, Set<Tag> tagsToDelete) {
+        requireNonNull(personToEdit);
+        requireNonNull(tagsToDelete);
+
+        Set<Tag> newTags = new HashSet<>(personToEdit.getTags());
+        newTags.removeAll(tagsToDelete);
+        return new Person(
+                personToEdit.getName(),
+                personToEdit.getPhone(),
+                personToEdit.getEmail(),
+                personToEdit.getAddress(),
+                newTags);
+    }
+
+    /**
+     * Determines what the CommandResult string should be.
+     * Handles cases of missing tags.
+     */
+    public static String formatResultMessage(Person personToEdit, Person editedPerson, Set<Tag> tagsToDelete) {
+        Set<Tag> toDelete = new HashSet<>(personToEdit.getTags());
+        toDelete.retainAll(tagsToDelete);
+
+        Set<Tag> missing = new HashSet<>(tagsToDelete);
+        missing.removeAll(toDelete);
```

Comment:
Perhaps this part could have comments describing what each variable is.
```suggestion
    public static String formatResultMessage(Person personToEdit, Person editedPerson, Set<Tag> tagsToDelete) {
        // toDelete contains the tags that can be deleted from the Person object
        Set<Tag> toDelete = new HashSet<>(personToEdit.getTags());
        toDelete.retainAll(tagsToDelete);

        // missing contains the tags that cannot be deleted from the Person object
        // (as they do not exist in the Person object)
        Set<Tag> missing = new HashSet<>(tagsToDelete);
        missing.removeAll(toDelete);
```

Reply:
```suggestion
    public static String formatResultMessage(Person personToEdit, Person editedPerson, Set<Tag> tagsToDelete) {
        Set<Tag> matchingTags = new HashSet<>(personToEdit.getTags());
        matchingTags.retainAll(tagsToDelete);

        Set<Tag> missingTags = new HashSet<>(tagsToDelete);
        missingTags.removeAll(matchingTags);
```

@jiakai-17 How about something like this instead? I'm generally not a fan of adding comments if the variable name suffices.